src/test/java/org/mastodon/mamut/feature/branch/exampleGraph/ExampleGraph3.java

```diff
@@ -0,0 +1,100 @@
+
+import org.mastodon.mamut.model.ModelGraph;
+import org.mastodon.mamut.model.Spot;
+import org.mastodon.mamut.model.branch.ModelBranchGraph;
+
+/**
+ * Represents a {@link AbstractExampleGraph} with the following {@link ModelGraph} and {@link ModelBranchGraph}:
+ *
+ * <h1>Spot-Graph</h1>
+ * <pre>
+ *                                                Spot( 0, X=1,00, Y=2,00, Z=3,00, tp=0 )
+ *                                                                   │
+ *                                                                   │
+ *                                                Spot( 1, X=2,00, Y=4,00, Z=6,00, tp=1 )
+ *                                                                   │
+ *                                                                   │
+ *                                                Spot( 2, X=3,00, Y=6,00, Z=9,00, tp=2 )
+ *                       ┌───────────────────────────────────────────┴──────────────────────┐
+ *                       │                                                                  │
+ *  Spot( 11, X=12,00, Y=24,00, Z=36,00, tp=3 )                         Spot( 3, X=4,00, Y=8,00, Z=12,00, tp=3 )
+ *                       │                                                                  │
+ *                       │                                                                  │
+ *                       │                                              Spot( 4, X=5,00, Y=10,00, Z=15,00, tp=4 )
+ *                       │                                            ┌─────────────────────┴─────────────────────┐
+ *                       │                                            │                                           │
+ *  Spot( 13, X=14,00, Y=28,00, Z=42,00, tp=5 )   Spot( 8, X=9,00, Y=18,00, Z=27,00, tp=5 )   Spot( 5, X=6,00, Y=12,00, Z=18,00, tp=5 )
+ *                                                                    │                                           │
+ *                                                                    │                                           │
+ *                                                                    │                       Spot( 6, X=7,00, Y=14,00, Z=21,00, tp=6 )
+ *                                                                    │                                           │
+ *                                                                    │                                           │
+ *                                               Spot( 10, X=11,00, Y=22,00, Z=33,00, tp=7 )  Spot( 7, X=8,00, Y=16,00, Z=24,00, tp=7 )
+ * </pre>
+ * <h1>BranchSpot-Graph</h1>
+ * <pre>
+ *                        branchSpot0
+ * 	       ┌──────────────┴─────────────────┐
+ * 	       │                                │
+ * 	   branchspot1                      branchSpot2
+ * 	┌──────┴───────┐
+ * 	│              │
+ * branchspot3 branchSpot4
+ * </pre>
+ */
+public class ExampleGraph3 extends AbstractExampleGraph
```

Comment:
I cannot spot the difference between ExampleGraph1 and ExampleGraph3. Please describe the difference in the javadoc.

A side note: The smaller the graphs, the easier it is to understand the unit tests. So please keep the small if possible.

Reply:
ExampleGraph3 has some Spots less than ExampleGraph1.

I would not like to describe differences between individual example graphs in their description. This would mean that in each example graph the differences to all other graphs would need to be described. I would argue that the existing Javadoc of the Spot-Graph and BranchSpot-Graph should be sufficient to find out the properties of an example graph.

Reply:
It is obvious that ExampleGraph3 is a copy of ExampleGraph1. But by reading the code, it's hard to spot the difference. Both graphs appear to be identical until I you use diff tool. And requiring a diff tool to read code is clearly an indication of hard to read code. There are two options to solve this problem:

1. Document the differences between ExampleGraph1, ExampleGraph3 and ExampleGraph4 as they are clearly copies of each other. The author knows that these examples are copies and where and why they are different. The javadoc is there to share this knowledge with the reader.

2. **But there is another option, which is much better:** Simplify ExampleGraph1. In order to achieve to goal of your test the graph only needs two spots: Spot11 and Spot13.  Is any of the other spots necessary?

Reply:
I decided to keep ExampleGraph4 and the 2 very similar ExampleGraph1 and ExampleGraph3. This should resolve the code redundancy issue.